aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -53,11 +50,12 @@ void bernoulli_mkl_kernel(Tensor &output, const double p, Generator* gen) {
 }
 #else
 void bernoulli_mkl_kernel(Tensor &self, const double p, Generator* gen) {
-  THGenerator* generator = get_generator(gen);
+  CPUGenerator* generator = check_generator_with_default<CPUGenerator>(gen, detail::getDefaultCPUGenerator().get());
+  // See Note [Thread-safety and Generators]
+  std::lock_guard<std::mutex> lock(generator->mutex_);
```

Comment:
Why did this lock guard migrate out of the block scope? It doesn't look like we're using generator afterwards, in which case it is better to keep the lock scope guarded.